### PR TITLE
Add winit feature flag

### PR DESCRIPTION
**Pull request by wusyong**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [x] Yes, tao and winit feature flags are added.
- [ ] No

### Checklist
- [ ] This PR will resolve #___
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary
- [x] It can be built on all targets and pass CI/CD.

### Other information
I want to add winit support for a very long time. Make wry to extend it and support Windows and macOS isn't that difficult.
I'm also working on [winit gtk](https://github.com/wusyong/winit-gtk) port which should make Linux support easier in the future.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -89,3 +89,10 @@ jobs:
           !contains(matrix.platform.target, 'android') &&
           !contains(matrix.platform.target, 'ios'))
         run: cargo test --verbose --target ${{ matrix.platform.target }}
+
+      - name: build wry with winit
+        if: (
+          contains(matrix.platform.target, 'windows') ||
+          contains(matrix.platform.target, 'apple') ||
```
Comment (amrbashir) [inline .github/workflows/build.yml]:
```suggestion
        if: (
          contains(matrix.platform.target, 'windows') ||
          contains(matrix.platform.target, 'apple'))
```

File: src/webview/webview2/mod.rs [hunk 2]
```diff
@@ -392,6 +390,8 @@ window.addEventListener('mousemove', (e) => window.chrome.webview.postMessage('_
             let mut js = PWSTR::null();
             args.TryGetWebMessageAsString(&mut js)?;
             let js = take_pwstr(js);
+            // Winit 0.28 doesn't have manual hittest yet.
+            #[cfg(not(feature = "winit"))]
```
Comment (amrbashir) [inline src/webview/webview2/mod.rs]: we can just port the code of hittest here, it shouldn't be a problem

**Review by amrbashir — CHANGES_REQUESTED** [review]
This is a fine way to add winit, but if there is time, we could maybe look into https://github.com/tauri-apps/wry/issues/677

File: src/webview/webview2/mod.rs (hunk 2 above)
Comment (wusyong) [reply src/webview/webview2/mod.rs]: There's also `window.begin_resize_drag`. And the hit_test function will need to call several DPI[ foreign functions](https://github.com/tauri-apps/tao/blob/dev/src/platform_impl/windows/util.rs#L318) which aren't included in windows crate.
Can you open an issue instead for future improvement?

**Review by amrbashir — DISMISSED** [review]

**Review by amrbashir — DISMISSED** [review]

**Review by lucasfernog — DISMISSED** [review]

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**